src/plugins/visualize/public/application/components/visualize_top_nav.tsx

```diff
@@ -165,14 +165,23 @@ const TopNav = ({
   ]);
 
   useEffect(() => {
+    const asyncSetIndexPattern = async () => {
+      let index: IndexPattern | null = null;
+
+      if (vis.type.getUsedIndexPattern) {
```

Comment:
@ppisljar maybe we should rethink that comment https://github.com/elastic/kibana/pull/82964#discussion_r521511096  and move getUsedIndexPattern into `vis.ts`
Now we know that we need that code at least from 2 different places?